Infrastructure/Services/ProcessEmailsService.cs

```diff
@@ -0,0 +1,450 @@
+                    if (throwExceptions)
+                    {
+                        await UpdateSentEmailsAsync(emailsToSend, notifyEmails);
+                        throw;
+                    }
+                }
+
+                emailsSent = await UpdateSentEmailsAsync(emailsToSend, notifyEmails);
+            }
+        }
+        else
+        {
+            await SetEmailsToDeactivatedEmailsAsync();
+        }
+
+        return emailsSent;
+    }
+
+    private async Task<int> UpdateSentEmailsAsync(IEnumerable<EmailLog> emailsToUpdate, IEnumerable<NotifyEmailDto> emailResults)
+    {
+        NotifyResponseMappingConfig.Configure();
+
+        var sentEmails = 0;
+        var updateSentDateTime = DateTime.UtcNow;
+
+        foreach (var emailResult in emailResults)
+        {
+            EmailLog? emailToUpdate = null;
+            try
+            {
+                emailToUpdate = emailsToUpdate.Single(x => x.ClientReferenceNumber == emailResult.ClientReference);
+
+                var notifyResponse = emailToUpdate.EmailNotifyResponseLog == null ? new EmailNotifyResponseLog() : emailToUpdate.EmailNotifyResponseLog;
+
+                notifyResponse = emailResult.NotifyResponse.Adapt(notifyResponse);
+
+                emailToUpdate.EmailNotifyResponseLog = notifyResponse;
+
+                var oldStatus = emailToUpdate.EmailStatusId;
+
+                if (!string.IsNullOrEmpty(notifyResponse.NotifyId))
+                {
+                    emailToUpdate.EmailStatusId = (int)EmailStatus.BeenProcessed;
+                    sentEmails++;
+                }
+                else
+                {
+                    emailToUpdate.EmailStatusId = (int)EmailStatus.ProcessingFailure;
+                    var retryInSeconds = _unitOfWork.EmailStatusRepository.GetById(emailToUpdate.EmailStatusId).RetrySendInSeconds;
+                    emailToUpdate.ProcessFromDate = updateSentDateTime.AddSeconds((double)retryInSeconds!);
+                }
+
+                emailToUpdate.LastEmailSendAttemptDate = updateSentDateTime;
+                if (oldStatus != emailToUpdate.EmailStatusId)
+                {
+                    emailToUpdate.LastStatusChangedDate = updateSentDateTime;
+                }
+
+                _unitOfWork.EmailLogRepository.Update(emailToUpdate);
+
+                await _unitOfWork.Complete();
+            }
+            catch (Exception ex)
+            {
+                //We don't want to throw the error here, we log and continue processing other emails
+                _logger.LogError(ex, "Unexpected error in UpdateSentEmailsAsync for EmailLog.ClientReferenceNumber: {ClientReferenceNumber}, EmailLog.Id: {EmailLogId}",
+                    emailToUpdate == null ? "NULL" : emailToUpdate.ClientReferenceNumber,
+                    emailToUpdate == null ? "NULL" : emailToUpdate.Id);
+            }
+        }
+
+        return sentEmails;
+    }
+
+    private async Task SetEmailsToDeactivatedEmailsAsync()
```

Comment:
The method `SetEmailsToDeactivatedEmailsAsync` could be renamed to `DeactivateEmailsAsync` for better clarity and consistency

Reply:
Done